syncer/syncer_test.go

```diff
@@ -1203,26 +1235,28 @@ func (s *testSyncerSuite) TestRun(c *C) {
 		jobNum := len(testJobs.jobs)
 		testJobs.RUnlock()
 
-		if jobNum >= len(testCases2) {
+		if jobNum >= len(expectJobs2) {
 			break
 		}
 	}
 
 	testJobs.RLock()
-	c.Assert(testJobs.jobs, HasLen, len(testCases2))
-	for i, testCase := range testCases2 {
-		c.Assert(testJobs.jobs[i].tp, Equals, testCase.tp)
-		if testJobs.jobs[i].tp == ddl {
-			c.Assert(testJobs.jobs[i].ddls[0], Equals, testCase.sqlInJob)
+	c.Assert(testJobs.jobs, HasLen, len(expectJobs2))
+	for i, job := range testJobs.jobs {
+		c.Log(i, job.tp, job.ddls, job.sql, job.args)
+
+		c.Assert(job.tp, Equals, expectJobs2[i].tp)
+		if job.tp == ddl {
+			c.Assert(job.ddls[0], Equals, expectJobs2[i].sqlInJob)
 		} else {
-			c.Assert(testJobs.jobs[i].sql, Equals, testCase.sqlInJob)
-			c.Assert(testJobs.jobs[i].args[0], Equals, testCase.arg)
+			c.Assert(job.sql, Equals, expectJobs2[i].sqlInJob)
+			c.Assert(job.args, DeepEquals, expectJobs2[i].args)
 		}
```

Comment:
It seems the execution of SQLs and the validation afterwards are almost identical for sql1 and sql2. Maybe we can extract a function or two to make the test simpler.

Reply:
updated https://github.com/pingcap/dm/pull/198/commits/47740ebba6850394b74fddb1558349c147e359b8

Reply:
Can we do the same for the execution and wait part?

Reply:
add a new function `executeSQLAndWait`